### PR TITLE
Remove panics based on team feedback

### PR DESCRIPTION
**Pull request by lsebrie**
Some panic where too noisy for incidents that shouldn't be so critical. When we improve our tools we could remove more of the ones we are keeping

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 234 issues
* ![MAJOR][MAJOR] 130 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 77 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L117](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L117): Constructor new co.rsk.core.bc.BlockChainImpl(RskSystemProperties, Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L201](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L201): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L203](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L203): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L315](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L315): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L374](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L374): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L385](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L385): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L390](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L390): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L471](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L471): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L552](https://github.com/rsksmart/rskj/blob/be753d3b98a2e36f99b683975d3c3ea6a90b7120/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L552): Method co.rsk.core.bc.BlockChainImpl.flushData() orders expressions in a conditional in a sub optimal way [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASEO_SUBOPTIMAL_EXPRESSION_ORDER)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/vm/VM.java [hunk 1]
```diff
@@ -1915,8 +1915,7 @@ public void steps(Program aprogram, long steps) {
                 vmCounter++;
             } // for
         } catch (RuntimeException e) {
-                logger.warn("VM halted: [{}]", e);
-                panicProcessor.panic("vm", String.format("VM halted: [%s]", e.getMessage()));
+                logger.error("VM halted: [{}]", e);
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/vm/VM.java]: Can you verify this is working correctly? I think the string formatting doesn't apply when the second parameter is of type `Exception`.

Comment (lsebrie) [reply rskj-core/src/main/java/org/ethereum/vm/VM.java]: :+1:

**Comment by rskops** [comment]
SonarQube analysis reported 221 issues
* ![MAJOR][MAJOR] 127 major
* ![MINOR][MINOR] 17 minor
* ![INFO][INFO] 77 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L117](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L117): Constructor new co.rsk.core.bc.BlockChainImpl(RskSystemProperties, Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L201](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L201): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L203](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L203): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L315](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L315): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L374](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L374): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L385](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L385): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L390](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L390): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L471](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L471): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L552](https://github.com/rsksmart/rskj/blob/fe45d86a858ca3898b96a7c345140e7424ff02a8/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L552): Method co.rsk.core.bc.BlockChainImpl.flushData() orders expressions in a conditional in a sub optimal way [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASEO_SUBOPTIMAL_EXPRESSION_ORDER)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 223 issues
* ![MAJOR][MAJOR] 129 major
* ![MINOR][MINOR] 17 minor
* ![INFO][INFO] 77 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L117](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L117): Constructor new co.rsk.core.bc.BlockChainImpl(RskSystemProperties, Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L201](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L201): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L203](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L203): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L315](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L315): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L374](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L374): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L385](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L385): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L390](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L390): Return an empty collection instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L471](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L471): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L552](https://github.com/rsksmart/rskj/blob/20304582ecb3442c6ddaa69df793c39250ada896/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L552): Method co.rsk.core.bc.BlockChainImpl.flushData() orders expressions in a conditional in a sub optimal way [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASEO_SUBOPTIMAL_EXPRESSION_ORDER)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**